Add OptionButton control

promag opened the pull request:
![image](https://user-images.githubusercontent.com/3534524/139280887-12a43ba7-929f-403b-a4ff-dd84e48a16bc.png)

[![Windows](https://svgshare.com/i/ZhY.svg)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/Win64%20\[unit%20tests,%20no%20gui%20tests,%20no%20boost::process,%20no%20functional%20tests\]%20\[focal\]/insecure_win_gui.zip?branch=pull/67)
[![macOS](https://svgshare.com/i/ZjP.svg)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macOS%2010.15%20\[gui,%20no%20tests\]%20\[focal\]/insecure_mac_gui.zip?branch=pull/67)

hebasto reviewed (commented):
Tested 7cde50b6a144ad87f6d05dfa100f9ecb29974b86 on Linux Mint 20.2 (Qt 5.12.8).

`Tab` (focus moving) and `Space` (focus selecting) work as expected.

What should be a role of the `Return` key in focus treatment?

Also:
```
$ src/qt/bitcoin-qt -signet
qrc:/qml/pages/stub.qml:34:9: QML Pane: Binding loop detected for property "implicitHeight"
qrc:/qml/pages/stub.qml:34:9: QML Pane: Binding loop detected for property "implicitHeight"
```

promag commented:
Thanks @hebasto, will address both comments.

promag commented on src/qml/controls/OptionButton.qml:
```diff
@@ -0,0 +1,73 @@
+        radius: 10
+        color: "transparent"
+        Rectangle {
+            visible: button.activeFocus
```
Will use `visualFocus` instead, see https://doc.qt.io/qt-5/qml-qtquick-controls2-control.html#visualFocus-prop.

promag commented:
Rebased and fixed warning. Now using `visualFocus` instead of `activeFocus`. Clicking or touching doesn't show focus border, only if the keyboard is used.

hebasto commented on src/qml/bitcoin_qml.qrc:
```diff
@@ -1,6 +1,8 @@
 <!DOCTYPE RCC><RCC version="1.0">
     <qresource prefix="/qml">
+        <file>controls/OptionButton.qml</file>
```
dc30b7085be0fa58e568515c730d2a7365c32d63

Sort lexicographically?

promag replied:
Done.

hebasto reviewed (approved):
ACK 8ab1626be7e717c6172582a89f70346481166778

hebasto reviewed (approved):
re-ACK 61668e989adc3fc53431665fddf4fd1f19926391

jarolrod commented on src/qml/pages/stub.qml:
```diff
@@ -5,33 +5,35 @@
     id: appWindow
     title: "Bitcoin Core TnG"
     minimumWidth: 750
     minimumHeight: 450
```
Do we have any plan on how to set the `minimumHeight` to correspond to the required minimum size of the contents?
<img width="862" alt="Screen Shot 2021-11-02 at 7 27 24 PM" src="https://user-images.githubusercontent.com/23396902/139965582-25901603-9537-4f5a-829f-154eb73b5668.png">

promag replied:
We should set these once we settle down the main/window layout.

jarolrod reviewed (commented):
ACK 61668e9

pending on ACK from our fellow designers. Pinging @Bosch-0 @GBKS

Bosch-0 commented:
ACK, LGTM on windows, this will be siloed to the onboarding / settings sections at a later stage correct?
